src/openforms/js/components/formio_builder/builder.js

```diff
@@ -411,12 +391,43 @@ const FormIOBuilder = ({
       prevLiteral,
       newLiteral
     );
-    exposeTranslations(newTranslations);
-    // update the literal for the next change cycle
     set(previousLiteralsRef.current, [changedPropertyPath], newLiteral);
+    return newTranslations;
   };
+
   // otherwise builder keeps refreshing/remounting
-  builderOptions.onChange = onBuilderFormChange;
+  builderOptions.onChange = (flags, changed, modifiedByHuman) => {
+    const {instance} = changed;
+
+    if (!flags.fromSubmission) {
+      instance.root.triggerChange(flags, changed, modifiedByHuman);
+    }
+
+    // we only need to sync translations when an actual human changed some literal
+    if (!modifiedByHuman) {
+      return;
+    }
```

Comment:
partially fixes some of the infinite change event triggers, but in generally this seems a bit of a safer approach - we only translate literals that are entered by the form designer, and don't even bother with automatically updated 'literals' since those aren't translated anyway.

Aborting here reduces the risk of change trigger cascades.

Reply:
this broke the translations update completely again so I removed it :grimacing: